src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/GreaterThanThresholdToVisibilityConverter.cs

```diff
@@ -9,16 +9,20 @@
 
 namespace NuGet.PackageManagement.UI
 {
-    public class DownloadCountToVisibilityConverter : IValueConverter
+    public class GreaterThanThresholdToVisibilityConverter : IValueConverter
     {
         public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
         {
             if (targetType == typeof(Visibility))
             {
-                long? downloadCount = value as long?;
-                if (downloadCount >= 0)
+                if (value is not null)
                 {
-                    return Visibility.Visible;
+                    long intValue = System.Convert.ToInt64(value, culture);
```

Comment:
Convert.ToInt64() will throw an exception if the value is not a valid long value. The example in the [doc](https://learn.microsoft.com/dotnet/api/system.convert.toint64?view=net-8.0#system-convert-toint64(system-object)) has catch blocks for various exceptions this method can throw.

Reply:
I'm guessing this is done so an object or a string can be converted to the correct value. Would it worth creating a `bool TryConvertToInt64(obj, formatter, out long? convertedValue)` method somewhere?

Reply:
@kartheekp-ms thanks for the feedback. @jgonz120 I like this suggestion. I've got this ready and will push changes in a moment.

Reply:
@kartheekp-ms @jgonz120 pushed an update that addresses this case and has a new test for invalid conversion scenarios.